src/DomainHost.sol

```diff
@@ -306,17 +306,45 @@ abstract contract DomainHost {
         emit Deficit(wad);
     }
 
+    /// @notice Governance is accruing surplus
+    /// @param debt The amount of debt in the guest domain that should be backed by pre-minted DAI [WAD]
+    function accrue(uint256 debt) external auth returns (uint256 _wad) {
```

Comment:
The idea would be Governance checks how much is the `debt` in L2 and pass it to generate the necessary `grain`, which will make sure enough DAI is in the escrow to cover the whole sub domain circulating supply.

Reply:
Anyway I'm very dubious about this approach. I guess it is probably better just to remove any message passing that something is not enforced or actually enforce it saving in storage.